src/wp-includes/media.php

```diff
@@ -1714,6 +1722,12 @@ function wp_img_tag_add_loading_attr( $image, $context ) {
 			$value = 'lazy';
 		}
 
+		// Images should have dimension attributes for the `loading` attribute
+		// to be added.
+		if ( false === strpos( $image, ' width=' ) || false === strpos( $image, ' height=' ) ) {
```

Comment:
I remember discussing this, I believe the consensus there was that this is unnecessary and be slightly overkill since this logic would run on image tags generated by WordPress (e.g. in post content) which would always use spaces only.